Python/specialize.c

```diff
@@ -667,14 +662,28 @@ _Py_Specialize_LoadAttr(PyObject *owner, _Py_CODEUNIT *instr, PyObject *name)
         }
         goto success;
     }
+    if (PyType_Check(owner)) {
+        int err = specialize_class_load_attr(owner, instr, name);
+        if (err) {
+            goto fail;
+        }
+        goto success;
+    }
+    int oparg = _Py_OPARG(*instr);
     PyTypeObject *type = Py_TYPE(owner);
     if (type->tp_dict == NULL) {
         if (PyType_Ready(type) < 0) {
             return -1;
         }
     }
-    PyObject *descr;
+    PyObject *descr = NULL;
     DescriptorClassification kind = analyze_descriptor(type, name, &descr, 0);
+    assert(descr != NULL || kind == ABSENT || kind == GETSET_OVERRIDDEN);
+    if ((oparg & 1) && kind == METHOD) {
+        if (specialize_attr_loadmethod(owner, instr, name, descr, kind)) {
```

Comment:
I think this should be handled in the switch on `kind` below, to make the code more regular.